diff --git a/sisl/linalg/base.py b/sisl/linalg/base.py
index c2ddb578..8e10d480 100644
--- a/sisl/linalg/base.py
+++ b/sisl/linalg/base.py
@@ -9,7 +9,7 @@ import numpy as np
 from numpy import atleast_1d, atleast_2d
 from scipy.linalg.blas import get_blas_funcs
 from scipy.linalg.lapack import get_lapack_funcs
-from scipy.linalg.misc import LinAlgError, _datacopied
+from scipy.linalg.misc import LinAlgError
 from scipy._lib._util import _asarray_validated
 import scipy.linalg as sl
 import scipy.sparse.linalg as ssl
@@ -20,6 +20,19 @@ from sisl._internal import set_module
 __all__ = ['linalg_info']
 
 
+def _datacopied(arr, original):
+    """
+    Strict check for `arr` not sharing any data with `original`,
+    under the assumption that arr = asarray(original)
+
+    """
+    if arr is original:
+        return False
+    if not isinstance(original, np.ndarray) and hasattr(original, '__array__'):
+        return False
+    return arr.base is None
+
+
 # Placeholder for basic linear algebra methods
 # I.e. when fetching the same method over and over
 # we should be able to reduce the overhead by retrieving the intrinsic version.
