diff --git a/sisl/_environ.py b/sisl/_environ.py
index f34d7729a..9c1c0c7f6 100644
--- a/sisl/_environ.py
+++ b/sisl/_environ.py
@@ -69,7 +69,12 @@ def get_environ_variable(name):
 
 
 # We register a few variables that may be used several places
-register_environ_variable("SISL_NUM_PROCS", min(1, len(os.sched_getaffinity(0))),
+try:
+    _nprocs = len(os.sched_getaffinity(0))
+except:
+    _nprocs = 1
+
+register_environ_variable("SISL_NUM_PROCS", min(1, _nprocs),
                           "Maximum number of CPU's used for parallel computing",
                           process=int)
 
diff --git a/sisl/viz/__init__.py b/sisl/viz/__init__.py
index 56ca2875f..76c25b519 100644
--- a/sisl/viz/__init__.py
+++ b/sisl/viz/__init__.py
@@ -19,7 +19,12 @@ The plotly backend.
 import os
 from sisl._environ import register_environ_variable
 
-register_environ_variable("SISL_VIZ_NUM_PROCS", min(1, len(os.sched_getaffinity(0))),
+try:
+    _nprocs = len(os.sched_getaffinity(0))
+except:
+    _nprocs = 1
+
+register_environ_variable("SISL_VIZ_NUM_PROCS", min(1, _nprocs),
                           description="Maximum number of processors used for parallel plotting",
                           process=int)
 
