diff --git a/sisl/viz/plots/bands.py b/sisl/viz/plots/bands.py
index 45c17ecb4..1602384ab 100644
--- a/sisl/viz/plots/bands.py
+++ b/sisl/viz/plots/bands.py
@@ -6,7 +6,10 @@ from functools import partial
 import itertools
 
 import numpy as np
-import xarray as xr
+try:
+    import xarray as xr
+except ModuleNotFoundError:
+    pass
 
 import sisl
 from sisl.physics.brillouinzone import BrillouinZone
